### DISCUSSION TITLE
Option to disable IPv6 support

### DISCUSSION THREAD

**Discussion by TheWebMachine** [music-assistant/support#2312] [Feature requests and ideas]
Any chance an option can be added to either Chromecast plugin or MA, at large, to completely disable IPv6 support?

I've been seeing a lot of heartbeat and other timeout errors with Google speakers and they are all IPv6 addresses in the message.

Mind you, I don't use IPv6 internally and streams play to speakers via IPv4, but a timeout to one of the IPv6 addresses sometimes pauses the stream with an "external source active" status. 

I want to effectively disable all IPv6 support in MA so it only uses IPv4, as attempted to use IPv6 may be contributing to some of my issues.

**Comment by TheWebMachine** [comment]
I even have IPv6 disabled in HA, but this obviously has no affect with add-on containers.

**Reply by marcelveldt** [reply]
Its 2024, you should enable IPv6 and not disable it, I truly do not understand what people have against IPv6. It brings a lot of good stuff. Heck, some services like Netflix will even slowdown if you still access it through IPv4.

**Reply by marcelveldt** [reply]
If the stream is paused while trying to access an IP that I consider a real bug

**Reply by TheWebMachine** [reply]
Well, I don't have an issue with IPv6 and use it in most of the rest of my network, but *not* the IoT side of things. You'll have to speak to the devices in my home that still don't support it properly...namely Google's damn speakers. 🤦🏻‍♂️ I had to ditch IPv6 for casting long before I started using MA, unfortunately. I really am getting disenchanted with my speakers, which is annoying because I have so many/so much money invested in them and they *used to* work great. 🤬

Anyway, what I'm seeing (which isn't much) is that while I'm casting to a speaker or group, if a timeout on an attempt to connect to the playing speaker via IPv6 happens, it causes the stream to stop and the player in MA says the speaker is playing an external source, but the player entity in HA still shows MA connected to it, as does my Android phone and Google Home app. 

Let me get it to happen again and catch as much log as I can. I'm still having an issue accessing the logs from within MA because the link it redirects to for the add-on is invalid from outside HA Supervisor. So, I can only grab snips from HA Add-On Logs tab, which is not ideal. 

I'll dig into it this weekend, but a toggle for IPv6 for even diagnostic purposes would be helpful to eliminate causes. Plus, as much as you and I would love to IPv6 all of the things (network engineer of 25yrs here), it just isn't possible until the manufacturers are forced to properly adopt it. Hell, even the Cisco RV345 I used to use had a sloppy IPv6 stack that only half worked. 🤷🏻‍♂️

**Reply by TheWebMachine** [reply]
And yes, I did try enabling the external access to the MA interface, tried getting log from there and still got an invalid page.

**Reply by OzGav** [reply]
Are you aware you can grab the full MA log from MA settings>>core

**Reply by TheWebMachine** [reply]
That is what I'm talking about. I get a 401 Unauthorized no matter how I try to get at it from Core.

![Screenshot_20240516_202519_Chrome](https://github.com/music-assistant/support/assets/1045052/6c304599-7c86-472a-aca6-510f6ae25946)

**Reply by OzGav** [reply]
Ok that’s weird. Does the rest of that URL look something like api/hassio_ingress/IygKyuk-B8Fp3hGwLvqoe5Q35ae_lplu-z_wM/log

and you don’t need the external port for that access it is available via ingress

**Reply by TheWebMachine** [reply]
You are correct about the format. The port is HA's port. I've also tried 80 and 443. I've tried everything I can think of. It's a normal HA OVA with MA running within an add-on container. The IP is the direct IP of the HA server. I have no idea why it doesn't work. I have previously completely removed and reinstalled MA to address other issues and it has no effect on this, so I don't know what else to try. 🤷🏻‍♂️

**Reply by OzGav** [reply]
If it has never worked then open an issue so Marcel can comment at some point
